cloud_snitch/migrate/uuid.py

```diff
@@ -0,0 +1,321 @@
+
+    Creates a query to drop the account number name constraint.
+    Creates a query to remove the account number name properties.
+    """
+    yield (
+        "DROP CONSTRAINT ON (environment:Environment) "
+        "   ASSERT environment.account_number_name IS UNIQUE;"
+    )
+    yield (
+        "DROP CONSTRAINT ON (environmentlock:EnvironmentLock) "
+        "   ASSERT environmentlock.account_number_name IS UNIQUE;"
+    )
+    yield "MATCH (e:Environment) SET e.account_number_name = NULL"
+    yield "MATCH (e:EnvironmentLock) SET e.account_number_name = NULL"
+
+
+def add_state():
+    """Generate query to add the first environment state node
+
+    Should only occur for environments that do not already have state.
+    The from property of the HAS_STATE relationship should be the environment's
+    created_at property. The to property should be the java end of time.
+
+    The last_synced property of the state should by the created_at property.
+    The status property should be 'migrated for uuid'.
+
+    :returns: Cipher query to add state to stateless environments.
+    :rtype: str
+    """
+    cipher = """
+        MATCH (e:Environment) WHERE NOT (e)-[:HAS_STATE]-()
+        CREATE (e)
+            -[r:HAS_STATE {{to: {}, from: e.created_at}}]
+            ->(firstState:EnvironmentState {{
+                last_sync: e.created_at,
+                status: 'migrated_for_uuid'
+            }})
+    """
+    cipher = cipher.format(EOT)
+    return cipher
+
+
+def execute(query):
+    """Executes an identity change query.
+
+    :param query: Instance of an identity change query
+    :type query: IdentityChangeQuery
+    """
+    driver = get_neo4j()
+    cipher = str(query)
+
+    with driver.session() as session:
+        with session.begin_transaction() as tx:
+            start = time.time()
+            summary = tx.run(cipher).summary()
+            changed = summary.counters.properties_set
+            print(
+                "\nExecuted the following query:\n{}\n"
+                "Changed {} properties in {:.3f}s"
+                .format(str(query), changed, time.time() - start)
+            )
+            print("Summary: {}".format(summary.counters))
+
+
+if __name__ == '__main__':
+    parser = base_parser(
+        description=(
+            "Migration script to convert to environment uuids. "
+            "All environments must have uuids."
+        )
+    )
+
+    parser.add_argument(
+        '--limit',
+        type=int,
+        default=2000,
+        help="Number of rows to consider at a time."
+    )
+
+    parser.add_argument(
+        '--cipher-only',
+        action="store_true",
+        default=False,
+        help="Print only the cipher for migration."
+    )
+
+    args = parser.parse_args()
+    # Check that all environments have uuids before migrating
+    if not validate_environments():
+        print(
+            "There are environments with uuids. "
```

Comment:
Similar to validate_environments() comment. shouldn't this be "There are environments without uuids."?